python/ml4ir/applications/classification/model/classification_model.py

```diff
@@ -62,52 +63,69 @@ def evaluate(
         -----
         You can directly do a `model.evaluate()` only if the keras model is compiled
         """
+        # FIXME: We need to come to a consensus for computing in pandas or spark and how we plan to
+        #  do that. In the mean time we will stick with just the model evaluation metrics
+        compute_group_metrics_in_memory = False
+
         if not self.is_compiled:
             return NotImplementedError
         group_metrics_keys = self.feature_config.get_group_metrics_keys()
         metrics_dict = self.model.evaluate(test_dataset)
         metrics_dict = dict(zip(self.model.metrics_names, metrics_dict))
-        predictions = self.predict(test_dataset,
-                                   inference_signature=inference_signature,
-                                   additional_features=additional_features,
-                                   logs_dir=None,
-                                   logging_frequency=logging_frequency)
-        global_metrics = []  # group_name, metric, value
-        grouped_metrics = []
-        # instead of calculating measure with a single update_state (can result in a call with
-        # thousands of examples at once, we use the same batch size used during training.
-        # Helps prevent OOM issues.
-        batch_size = test_dataset._input_dataset._batch_size.numpy()  # Hacky way to get batch_size
-        # Letting metrics in the outer loop to avoid tracing
-        for metric in self.model.metrics:
-            global_metrics.append(self.calculate_metric_on_batch(metric, predictions, batch_size))
-            self.logger.info(f"Global metric {metric.name} completed."
-                             f" Score: {global_metrics[-1]['value']}")
-
-            for group_ in group_metrics_keys:  # Calculate metrics for group metrics
-                for name, group in predictions.groupby(group_['name']):
-                    self.logger.info(f"Per feature metric {metric.name}."
-                                     f" Feature: {group_['name']}, value: {name}")
-                    if group.shape[0] >= group_metrics_min_queries:
-                        grouped_metrics.append(self.calculate_metric_on_batch(metric,
-                                                                              group,
-                                                                              batch_size,
-                                                                              group_['name'], name))
-        global_metrics = pd.DataFrame(global_metrics)
-        grouped_metrics = pd.DataFrame(grouped_metrics).sort_values(by='size')
-        if logs_dir:
-            self.file_io.write_df(
-                grouped_metrics,
-                outfile=os.path.join(logs_dir, RelevanceModelConstants.GROUP_METRICS_CSV_FILE),
-                index=False
+        if compute_group_metrics_in_memory:
```

Comment:
Temporary patch till we decide on a plan for next quarter

Reply:
can we make this an argparser argument with default to `True` (for the huge majority of datasets I know, it would pass easily)?
Also, I would suggest adding just below: 
```
self.logger.info('Computing metric scores for the grouped metrics in the feature_config. Warning: currently, this collects the test data and predictions in memory in order to perform the groupBy operations. With very big data, it can lead in OOM issues.')
```
What do you think?

Reply:
Makes sense. Made the changes 👍